CMakeLists.txt

```diff
@@ -443,3 +473,13 @@ endif()
 if(OpenCL_FOUND)
   target_link_libraries( opmsimulators ${OpenCL_LIBRARIES} )
 endif()
+
+if(HAVE_FPGA)
+  add_dependencies(opmsimulators FPGA_library)
+  target_link_libraries( opmsimulators PUBLIC ${_modules_dir}/opm-simulators/opm/simulators/linalg/bda/fpga/alveo_u280/vitis_20192/OPM-integration/fpga_lib_alveo_u280.a )
+  # link Xilinx-specific OpenCL library if OpenCL not found
+  if (NOT OpenCL_FOUND)
+    # used here is the default installation path for the library, change as needed
+    target_link_libraries( opmsimulators PUBLIC /opt/xilinx/xrt/lib/libxilinxopencl.so )
+  endif()
```

Comment:
Can't we make CMake find the correct opencl implementation by setting the OCL_ROOT environment variable or OPENCL_INCLUDE_DIR OPENCL_LIBRARY as a user?

What you are doing might only work on your personal system.

Reply:
After more testing, this library does not seem to be needed, hence the whole block has been removed.